### PR TITLE
Move from ref to branch names

### PR DESCRIPTION
**Pull request by elibon99**
This change:
 - Moves from reference to branch name (i.e. "main" instead of "refs/heads/main") matching
 - Allows verifying a specific commit by specifying a bootstrap.

### CODE REVIEW

File: gitbark/objects.py [hunk 1]
```diff
@@ -67,8 +67,8 @@ def get_default(cls, pattern: str, bootstrap: str) -> "BranchRule":
         return cls(pattern=pattern, bootstrap=bootstrap, ff_only=True)
 
     def branches(self, repo: Repository) -> list[str]:
-        pattern = re.compile(f".*{self.pattern}")
-        all_branches = list(repo.references)
+        pattern = re.compile(f"{self.pattern}")
```
Comment (dainnilsson) [inline gitbark/objects.py:70]: You can skip the f-string and just use self.pattern here

File: gitbark/cli/__main__.py [hunk 2]
```diff
@@ -178,53 +187,14 @@ def click_parse_bootstrap(ctx, param, val):
...
-    callback=click_parse_branch,
-)
-@click.option(
-    "-r",
-    "--ref-update",
-    type=(str, str, str),
-    help="The reference update.",
-    callback=click_parse_ref_update,
-)
+@click.argument("object", callback=click_parse_commit_or_branch, default="HEAD")
```
Comment (dainnilsson) [inline gitbark/cli/__main__.py:197]: Would "target" be a more suitable name?

**Review by dainnilsson — APPROVED** [review]

File: gitbark/objects.py (hunk 1 above)
Comment (elibon99) [reply gitbark/objects.py]: You're right! Pushed a commit that changes this.

File: gitbark/cli/__main__.py (hunk 2 above)
Comment (elibon99) [reply gitbark/cli/__main__.py]: I agree! The latest commit resolves this.

**Pull request merged**